cpp/src/arrow/meson.build

```diff
@@ -293,7 +293,10 @@ if needs_csv
 endif
 
 if needs_json or needs_integration
-    rapidjson_dep = dependency('rapidjson', include_type: 'system')
+    rapidjson_dep = dependency(
+        'RapidJSON',
```

Comment:
@kou thanks for pointing out https://github.com/apache/arrow/pull/47318#issuecomment-3186851513 - that was really helpful. This should fix the issue